### ISSUE TITLE
[Bug] Bug when use CrossFFN module and init_model function at the same time.

### ISSUE THREAD

**Issue by liupgd** [open-mmlab/mmpose#2448]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmpose/issues) and [Discussions](https://github.com/open-mmlab/mmpose/discussions) but cannot get the expected help.
- [X] The bug has not been fixed in the latest version(https://github.com/open-mmlab/mmpose).

### Environment

The bug is obvious, I'm using these versions. 
mmcv                          2.0.0
mmdet                         3.0.0rc5
mmengine                      0.7.0
mmpose                        1.0.0 

### Reproduces the problem - code sample

The `mmpose/apis/inference.py init_model` function will call `revert_sync_batchnorm`, where SyncBatchNorm will be replaced by ordinary batchnorm module recursively.   
However, in CrossFFN(mmpose/models/backbones/hrformer.py) module, the layers are just wrapped as list:
```python
class CrossFFN(BaseModule):
    ....
    self.layers = [
            self.fc1, self.norm1, self.act1, self.dw3x3, self.norm2, self.act2,
            self.fc2, self.norm3, self.act3
        ]
    def forward(...):
        ....
        for layer in self.layers:
            x = layer(x)
        ...
```
when run `revert_sync_batchnorm`, the `self.layers` member will not be replaced with new norm modules. When run forward, `self.layer[1]` is not the new  `self.norm1`. 
The direct effect is when calling `model.to("cuda")`, norms in `self.layers`  will remains in CPU RAM. A runtime error(https://github.com/open-mmlab/mmpose/issues/2236) will happen.

### Reproduces the problem - command or script

略

### Reproduces the problem - error message

略

**Comment by liupgd** [comment]
I hope official guys can check other model codes where vallina `list` is used in module definition. Think about replacing them with `nn.ModuleList`.

**Comment by Ben-Louis** [comment]
Thank you very much for your feedback. This bug has been fixed in #2449
